app/ide-desktop/lib/dashboard/src/modals/ManagePermissionsModal.tsx

```diff
@@ -44,8 +44,11 @@ const TYPE_SELECTOR_Y_OFFSET_PX = 32
 export interface ManagePermissionsModalProps<
   Asset extends backendModule.AnyAsset = backendModule.AnyAsset,
 > {
-  readonly backend: Backend
-  readonly item: Asset
+  readonly item: {
```

Comment:
similarly here re: nesting. perhaps these should be `Pick<backendModule.AnyAsset, 'a' | 'b' | 'c'>`?

Reply:
Questionable, on the one hand - this makes sense, on the other hand, it leads to unnecessary coupling between a backend and the modal. I think it might be a good idea to decouple these things as much as we can to be able to reuse this modal in mocked environments and other places. WDYT?

Reply:
while that makes sens, i think it does make sense to couple them *somewhat*, at least - otherwise you can no longer pass `asset`s directly to these functions, right? like, *technically* there is a dependency here. but to the user, they can't see the keys that have been picked away so it should make no difference. `backendModule` is almost all just type declarations anyway - there is one class, but it's an abstract class so i think it basically counts as a type as well. similarly with the constructors for the newtypes.

Reply:
tl;dr:
- as long as we `import type * as backendModule`, it shouldn't affect mocking without having to import the actual module at runtime
- we want it to be coupled so that we can pass `asset`s directly

worth noting that we *techincally* already do something similar when picking `pageX` and `pageY` or whatever from `MouseEvent` from certain fixed-positioned modals.

Reply:
> while that makes sense, i think it does make sense to couple them somewhat, at least - otherwise you can no longer pass assets directly to these functions, right?

I guess not, due to the duck-typing system in ts we should be able to pass anything that extends the declared shape, no matter whether explicitly or implicitly.

Reply:
So the real question is more like: do we _want_ to explicitly say that this modal should work only with assets, or with _anything_ that has these fields? Shall we use the same modal to assign permissions to assets, and also, maybe assign permissions to users to access certain pages, e.g. manage users/groups/subscriptions? 
It's a good question and I don't have an answer here

Reply:
>  we should be able to pass anything that extends the declared shape

exactly my point - i just mean that assets are the "first-class" users of this modal, right? i'm just thinking, if it goes out of sync with the assets then we'd have to do `{ foo2: asset.foo, bar: asset.bar }` or similar rather than the more intuitive `asset`?

or i guess iow: i think assets should be the "source of truth" of what the shape looks like - `Pick` omits all keys other than the relevant ones anyway, so as long as the keys are there the code will typecheck



but perhaps you mean it should be decoupled further? maybe we should be providing e.g. `setPermissions` as a prop? but i suppose it's probably not particularly relevant right now, and we can probably refactor later on when we're sure we need to use it for other things?

Reply:
Lets use pick then :)